File: 1232-check-if-it-is-a-straight-line/1232-check-if-it-is-a-straight-line.cpp
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {
        int diffX = coordinates[1][0] - coordinates[0][0];
        int diffY = coordinates[1][1] - coordinates[0][1];
        for(vector<int> coor: coordinates) {
            if(diffX * (coor[1] - coordinates[0][1]) != diffY * (coor[0] - coordinates[0][0])) return false;
        }
        return true;
    }
};

File: 1232-check-if-it-is-a-straight-line/1232. Check If It Is a Straight Line.go
func checkStraightLine(coordinates [][]int) bool {
    diffX := coordinates[1][0] - coordinates[0][0]
    diffY := coordinates[1][1] - coordinates[0][1]
    for _, coor := range coordinates {
        if diffX * (coor[1] - coordinates[0][1]) != diffY * (coor[0] - coordinates[0][0]) {return false}
    }
    return true
}
